### ISSUE TITLE
Bug: version 2025.0.1 breaking changes

### ISSUE THREAD

**Issue by leandrodamascena** [FHPythonUtils/LicenseCheck#110]
## Before You Begin

Before proceeding, please make sure to follow these steps:

- [x] I have checked for similar issues in the project's issue tracker.
- [x] I have searched closed issues to see if a similar problem was reported
  before.

## Issue Details

Hello! I'm using licensecheck to check dependencies in a poetry project and the command `licensecheck -u poetry:dev --zero` is not working anymore. What is the option to check required + dev dependencies in a poetry project using this new version?

Another thing, if I run `licensecheck --group dev --zero` it just check dev dependencies and a weird thing is happening: it marks `licensecheck` as an UNKNOW license!

```sh
(aws-lambda-powertools-py3.9) ➜  aws-lambda-powertools-v3 git:(dependabot/pip/develop/licensecheck-2025.0.1) ✗ pip list |grep licensecheck
licensecheck                                2025.0.1
```

```sh
(aws-lambda-powertools-py3.9) ➜  aws-lambda-powertools-v3 git:(dependabot/pip/develop/licensecheck-2025.0.1) ✗ licensecheck --group dev --zero |grep license
2025-03-18 09:45:29.703 | WARNING  | licensecheck.checker:resolve_requirements:33 - ('error: Requested extra not found: dev\n', '')
│ program         │ licensecheck │
│ license         │ MIT LICENSE  │
│ project_license │ MIT LICENSE  │
│ ✔          │ license-expression                          │ APACHE-2.0                            │
│ ✖          │ licensecheck                                │ UNKNOWN                            
```

### Expected Behavior

Old commands continues working.

### Actual Behavior

Not working

Thank you

## System Information

Please provide the following additional information about your system or
environment:

- Operating System (OS):
- OS Version:
- Browser (if applicable):
- Browser Version (if applicable):
- Project Version (if applicable):

**Comment by leandrodamascena** [comment]
Hello! I hope things are going well! Did you had a chance to take a look on this?

**Comment by FredHappyface** [comment]
Hi thanks for the nudge. Life is super busy atm. Yes there are breaking changes is new versions. I've attempted to document this vest I can for now, those this is wip

Please check 

https://github.com/FHPythonUtils/LicenseCheck?tab=readme-ov-file#examples-from-the-command-line

https://github.com/FHPythonUtils/LicenseCheck/blob/master/documentation/user/examples.md

And the output of licensecheck --help

**Comment by bombsimon** [comment]
Thanks for this tool and the documentation, great work!

Just curious since you're linking the docs, is this expected to work? I never had this working before but using `2025.1.0` now with `poetry` and we define dev dependencies under `[tool.poetry.group.dev.dependencies]`.

Running `licensecheck` works just fine with the new `uv` resolver. However, when I try `-g dev` as per this issue, I get this warning too, but it does seem to do _something_ different. The output now contains dev dependencies (such as `licensecheck` itself) but all packages are now printed in SCREAMING-KEBAB-CASE. Also some of the dependencies seen without the group are now gone - but not all.

Just to confirm, is this the expected behavior? I consulted the docs you linked but can't find a way to always include all group, is that possible? Lastly, would your recommended workaround be to swap out `poetry` to `uv` for best compatibility?

**Comment by FredHappyface** [comment]
Explicit poetry group support is added via the native resolver in https://github.com/FHPythonUtils/DepGather, note that this should in theory be supported by the uv resolver too. I'll create a task to create a test case for this and close this issue 

If you have the bandwidth for your project/product uv is well worth checking out but licensecheck is intended to support poetry to a reasonable degree (such as groups)
